src/shared/sleep-config.c

```diff
@@ -108,77 +111,162 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
+                        c++;
+        }
+        /* if all battery are below 5 then c will remain 0*/
+        if (c == 0)
+                return 1;
+        return 0;
 }
 
 /* Battery percentage capacity fetched from capacity file and if in range 0-100 then returned */
-int read_battery_capacity_percentage(void) {
-        _cleanup_free_ char *bat_cap = NULL;
+int read_battery_capacity_percentage(sd_device *dev) {
+        const char *x = NULL;
         int battery_capacity, r;
 
-        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
+        assert(dev);
+
+        r = sd_device_get_property_value(dev, "POWER_SUPPLY_CAPACITY", &x);
         if (r == -ENOENT)
-               return log_debug_errno(r, "/sys/class/power_supply/BAT0/capacity is unavailable. Assuming no battery exists: %m");
+                return log_device_debug_errno(dev, r, "battery capacity is unavailable. Assuming no battery exists: %m");
         if (r < 0)
-               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
+                return log_device_debug_errno(dev, r, "Failed to read battery capacity: %m");
 
-        r = safe_atoi(bat_cap, &battery_capacity);
+        r = safe_atoi(x, &battery_capacity);
         if (r < 0)
-               return log_debug_errno(r, "Failed to parse battery capacity: %m");
+                return log_debug_errno(r, "Failed to parse battery capacity: %m");
 
         if (battery_capacity < 0 || battery_capacity > 100)
-               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
+                return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
 
         return battery_capacity;
 }
 
+/* Store current capacity of each battery before suspension and timestamp */
+int store_battery_capacity(Hashmap *current_capacity) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        int battery_capacity, r;
+
+        assert(current_capacity);
+
+        current_capacity = hashmap_new(&string_hash_ops);
+        if (!current_capacity)
+                return log_oom();
+
+        r = get_sd_device_battery(&e);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get available battery: %m");
+
+        FOREACH_DEVICE(e, dev) { /* For AC, the TYPE is mains */
+                const char *x = NULL;
+
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get battery capacity: %m");
+                        continue;
+                }
+                battery_capacity = r;
+
+                r = sd_device_get_property_value(dev, "POWER_SUPPLY_NAME", &x);
+                if (r == -ENOENT)
+                        return log_device_debug_errno(dev, r, "battery capacity is unavailable. Assuming no battery exists: %m");
```

Comment:
this should log and continue

Reply:
you mean this should skip the current iteration if the battery name is not found. 
because if battery name does not exist, it would be safe to skip that iternation since we can not map battery capacity in that case with some random value.